lib/extensions/ar_taggable.rb

```diff
@@ -77,6 +77,38 @@ def tags(options = {})
       options[:ns] = Tag.get_namespace(options)
       Tag.tags(options)
     end
+
+    # defines an attribute that detects a certain tag
+    # define has_attrs, has_attrs? and attr_tags
+    # arel returns true or nil (false)
+    #   a count() > 0 or a case statement could return true, false (instead of null) but this seemed the least verbose
+    def tag_attribute(attribute_name, ns)
+      plural_attribute_name = "has_#{attribute_name.to_s.pluralize}"
+
+      virtual_attribute plural_attribute_name, :boolean, :uses => :tags, :arel => (lambda do |t|
+        ta = Tag.arel_table
+        tnga = Tagging.arel_table
+        t.grouping(
+          Arel.sql(
+            Tagging.joins(:tag).select('true')
+                   .where(ta[:name].matches("#{ns}/%", nil, true))
+                   .where(tnga[:taggable_type].eq(base_class.name).and(tnga[:taggable_id].eq(arel_table[:id])))
+                   .limit(1).to_sql
+          )
+        )
+      end)
+
+      define_method("#{attribute_name}_tags") do
+        Tag.filter_ns(tags, ns)
+      end
+
+      define_method(plural_attribute_name) do
+        has_attribute?(plural_attribute_name) ? self[plural_attribute_name] || false : Tag.filter_ns(tags, ns).size > 0
+      end
+
+      # Do we want to define "#{attribute_name}_tags" or #{attribute_name}
```

Comment:
Not sure why this question is here as a comment...seems more important as a review comment than as part of the source?